### PR TITLE
feat: make 2fa great again

### PR DESCRIPTION
**Pull request by zlwaterfield**
## Changes

This PR provides a bunch of 2fa improvements - follow up to #26526
- more consistent naming across variables, components, endpoints, etc.
- pull together all modals into a single reusable one
- improve styling of 2fa related ui
- allows backup codes to be used
- reduces back up codes down to 5

Questions
- what security implications should we be thinking about?
	- blocking delete 2fa and generate/re-generate backup codes with the time sensitive auth check
	- sending emails when 2fa added, remove, or backup code used. 

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

It doesn't have an impact

## How did you test this code?

Lots of manual testing + added tests

### CODE REVIEW

File: frontend/src/scenes/authentication/TwoFactorSetupModal.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    closable = true,
+    required = false,
+    forceOpen = false,
+}: TwoFactorSetupModalProps): JSX.Element {
+    const { isTwoFactorSetupModalOpen } = useValues(twoFactorLogic)
+    const { toggleTwoFactorSetupModal } = useActions(twoFactorLogic)
+
+    return (
+        <LemonModal
+            title="Set up 2FA"
```
Comment (Twixes) [inline frontend/src/scenes/authentication/TwoFactorSetupModal.tsx:26]: Not necessary to use an acronym here
```suggestion
            title="Set up two-factor authentication"
```

**Review by Twixes — APPROVED** [review]
It's great. Not even again, more for the first time

File: posthog/api/test/test_authentication.py [hunk 2]
```diff
@@ -322,6 +331,78 @@ def test_2fa_throttling(self):
...
+
+        # Use backup code once
+        response = self.client.post("/api/login/token", {"token": "123456"})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Log out
+        self.client.logout()
+
+        # Wait for throttling to expire
+        import time
```
Comment (benjackwhite) [inline posthog/api/test/test_authentication.py:374]: Weird place to import?

**Review by benjackwhite — APPROVED** [review]
One non blocking thing. This is awesome work.

**Pull request merged**